src/random_lcg.C

```diff
@@ -0,0 +1,144 @@
+#include "random_lcg.h"
+#include <cmath>
+
+
+// Constants
+extern "C" const int N_STREAMS         {5};
+extern "C" const int STREAM_TRACKING   {0};
+extern "C" const int STREAM_TALLIES    {1};
+extern "C" const int STREAM_SOURCE     {2};
+extern "C" const int STREAM_URR_PTABLE {3};
+extern "C" const int STREAM_VOLUME     {4};
+
+// Starting seed
+int64_t seed {1};
+
+// LCG parameters
+constexpr uint64_t prn_mult   {2806196910506780709LL};   // multiplication
+                                                         //   factor, g
+constexpr uint64_t prn_add    {1};                       // additive factor, c
+constexpr uint64_t prn_mod    {0x8000000000000000};      // 2^63
+constexpr uint64_t prn_mask   {0x7fffffffffffffff};      // 2^63 - 1
+constexpr uint64_t prn_stride {152917LL};                // stride between
+                                                         //   particles
+constexpr double   prn_norm   {pow(2, -63)};             // 2^-63
```

Comment:
The Intel compiler chokes on this because pow isn't considered a constexpr function. An easy way to get around this is to just rewrite it as 1./prn_mod.